File: lessons/compiled_lang/rust/src/e20.branch.rs
use std::io::{self, Write};

fn main() {
    print!("Input a number: ");
    io::stdout().flush().unwrap();

    let mut line = String::new();
    io::stdin().read_line(&mut line).unwrap();
    let number: i32 = line.trim_end().parse().unwrap();

    if number > 0 {
        println!("Number is greater than 0");
    } else if number < 0 {
        println!("Number is less than 0");
    } else {
        println!("Number is 0");
    }
}

File: lessons/compiled_lang/java/src/e20.branch.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class E20Branch {
    public static void main(String[] args) throws IOException {
        BufferedReader stdin = new BufferedReader(new InputStreamReader(System.in));
        System.out.print("Input a number: ");
        int number = Integer.parseInt(stdin.readLine());

        if (number > 0)
            System.out.println("Number is greater than 0");
        else if (number < 0)
            System.out.println("Number is less than 0");
        else
            System.out.println("Number is 0");
    }
}
